ReactAndroid/src/main/java/com/facebook/react/fabric/mounting/SurfaceMountingManager.java

```diff
@@ -992,9 +993,21 @@ public void updateLayout(int reactTag, int x, int y, int width, int height, int
       parent.requestLayout();
     }
 
-    // TODO: T31905686 Check if the parent of the view has to layout the view, or the child has
-    // to lay itself out. see NativeViewHierarchyManager.updateLayout
-    viewToUpdate.layout(x, y, x + width, y + height);
+    ViewState parentViewState = getViewState(parentTag);
+    ReactViewManagerWrapper parentViewManager = parentViewState.mViewManager;
+    IViewManagerWithChildren parentViewManagerWithChildren;
+    if (parentViewManager instanceof IViewManagerWithChildren) {
```

Comment:
This crashes in the new architecture with `com.facebook.react.uimanager.IllegalViewOperationException: Trying to use view with tag 1 as a parent, but its Manager doesn't implement IViewManagerWithChildren`

I believe this is because we don't have an associated ViewManager for the RootView in the new arch.

Reply:
The old architecture checked if it was a root tag. Is there an equivalent to that I can run in the new architecture?

https://github.com/facebook/react-native/blob/bd3cb7c83c411f6580b7049328834178d1cb6fa5/ReactAndroid/src/main/java/com/facebook/react/uimanager/NativeViewHierarchyManager.java#L208-L212

Reply:
I'd change this check to just ignore this condition if mViewManager is null.

Reply:
Thanks. I've added the check and tried to keep it as similar as possible to the equivalent check in the old architecture.